build-conflicts.php

```diff
@@ -199,6 +226,24 @@ static function (array $components): array {
         )),
     ));
 
+    $prevComposerJSONFileData = file_get_contents(__DIR__ . '/build/roave-security-advisories/composer.json');
+
+    /** @var array $prevComposerDecodedData
+     */
+    $prevComposerDecodedData  = json_decode($prevComposerJSONFileData, true);
+
+    /** @var array<array-key, string> $oldConflictPackages */
+    $oldConflictPackages      = array_keys($prevComposerDecodedData['conflict']);
+
+    $addedAdvisories = [];
+    foreach ($getAdvisories() as $advisory) {
+        if (in_array($advisory->package->packageName, $oldConflictPackages, true)) {
```

Comment:
Well, this would only list **new packages**, but not new versions of those packages :thinking: 

A bit of a more elaborate intersection mechanism is needed here, perhaps reading the current `composer.json` into a set of exclusion ranges (we already have value objects for that)

Reply:
oh, okay, thank you. I'll try to tackle it this weekend :thinking:

Reply:
No need to push it, don't worry! We can also try discussing this 1-on-1 in the coming weeks, if you want.

Reply:
...and I got overwhelmed with stuff. Sorry for late response. Yes, 1-on-1 would be great as I'm not sure I understand what is needed to be done. 
For example we have an old set of ranges `"amphp/http-client": "<1|>=4,<4.4" `
and new set of ranges that looks like so `"amphp/http-client": ">=4,<4.4|>=4.6"`, 
where `>=4.6` is new and `<1` was deleted/retracted. 

I should mention both `<1` and `>=4.6` in a commit message, right ? Basically everything that is not "overlapping" between the two :thinking: 

@azjezz thank you for suggestions! It was a bit hard to wrap my head around some PSL stuff.

Reply:
been staring at this whole evening. An elaborate intersection mechanism is, of course, possible, but looks overly complex(?) :thinking: 
If that is for human to read, could we just do something dumb like this:

- get old and fresh json files with conflicts
- if package is completely new, mark it as such
- if package exists in both files, try to compare string values.
- if strings are different, then add it, for example, like so 
```
"amphp/http-client"  updated from  "<1|>=4,<4.4" to  ">=4,<4.4|>=4.6"
```

And finally in commit message, we'd have it 
```
Newly added:
   Package: "amphp/http"  "<1.0.1"
   Summary: "Potential SQL injection vector"
   URI: "https://github.com/...."
 
Updated:
   "amphp/http-client"  updated from  "<1|>=4,<4.4" to  ">=4,<4.4|>=4.6"
   Summary: "Potential SQL injection vector"
   URI: "https://github.com/...."

Deleted (completely):
   "foo/bar" ">1.2"
   Summary: "Potential SQL injection vector"
   URI: "https://github.com/...."
```

Just thinking out loud :see_no_evil:

Reply:
> An elaborate intersection mechanism is, of course, possible, but looks overly complex(?) thinking

We already have that.

 1. read current range
 2. read advisory
 3. compare advisory with current range: if it is already **fully included**, then consider it "done"
 4. otherwise, add it to the bucket of "advisories that will change `composer.json`"

Reply:
Discussed briefly on twitter:

 1. read current `composer.json` into a `list<Component>`, where each `Component` has its associated dependency exclusion range parsed (we already have most of that written)
 2. read the individual advisories from their respective sources - keep a link to the source in each of the read advisories
 3. compare the `list<Component>` to the `list<Advisory>`, and determine which `Advisory` instances are **NOT** fully contained in `list<Component>`
 4. use the resulting data to produce the commit message